### PR TITLE
Add type-safe accessor for the logger

### PR DESCRIPTION
**Pull request by kegsay**

### CODE REVIEW

**Comment by NegativeMjark** [comment]
This looks good. Ideally we'd also hide the `ContextKeys` type so that other modules can't set the wrong type of logger into the context.

File: json.go [hunk 1]
```diff
@@ -18,6 +18,15 @@ type ContextKeys string
 // CtxValueLogger is the key to extract the logrus Logger.
 const CtxValueLogger = ContextKeys("logger")
```
Comment (NegativeMjark) [inline json.go]: Can we make the `ContextKeys` type and the `CtxValueLogger` const private?

**Review by NegativeMjark — CHANGES_REQUESTED** [review]
Yeah, I'd like us to hide the Context keys here.

File: json.go (hunk 1 above)
Comment (kegsay) [reply json.go]: Done.

**Comment by kegsay** [comment]
@NegativeMjark PTAL

**Review by NegativeMjark — APPROVED** [review]
LGTM

**Pull request merged**